Add clarifying comment for field parsing in TABLE_SHARE::init_from_binary_frm_image

Krishnateja4367 opened the pull request:
This small change adds a helpful comment above the loop that extracts column definitions
from the .frm image inside TABLE_SHARE::init_from_binary_frm_image().

It highlights how column metadata like names, types, and flags are handled.

This is part of my exploration of the frm file format for GSoC 2025 under MDEV-18827.  
I’d love any feedback or suggestions!

Thank you

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/MariaDB/server?pullRequest=3938) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/MariaDB/server?pullRequest=3938) before we can accept your contribution.<br/><hr/>**Krishna Teja** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/MariaDB/server?pullRequest=3938) it.</sub>

svoj commented:
@cvicentiu what is the status of #2191? Should we stick to #2191 and try to get it completed, or is this task good enough for another gsoc round?

Krishnateja4367 commented on sql/table.cc:
```diff
@@ -2473,6 +2473,9 @@ int TABLE_SHARE::init_from_binary_frm_image(THD *thd, bool write,
       field_data_type_info_array.parse(old_root, share->fields,
                                        extra2.field_data_type_info))
     goto err;
+// Extract column definitions from the .frm image buffer.
```
Hi, I've updated the comment to use a proper C-style block as suggested. Let me know if anything else needs to be changed. Thanks!

FooBarrior replied:
[CODING STANDARDS](https://github.com/MariaDB/server/blob/main/CODING_STANDARDS.md#comments) has nothing against using single-line comments, @DaveGosselin-MariaDB!

DaveGosselin-MariaDB replied:
Agreed, but this was a multiline comment 😄 (please see the original commit @FooBarrior ).  Maybe our coding standards also allow C++ style commenting for multiline comments?

DaveGosselin-MariaDB replied:
Just checked our coding standards and C style comments are preferred for block comments.

FooBarrior commented on sql/table.cc:
```diff
@@ -2473,6 +2473,9 @@ int TABLE_SHARE::init_from_binary_frm_image(THD *thd, bool write,
       field_data_type_info_array.parse(old_root, share->fields,
                                        extra2.field_data_type_info))
     goto err;
+  /*
```
The comment sticks to the wrong block: it's fine to have no space before the for loop, but not fine to have a space between for and a comment block, and at the same time no empty line in between goto and a comment.

Either way to fix this is fine.

FooBarrior commented on sql/table.cc:
```diff
@@ -2473,6 +2473,9 @@ int TABLE_SHARE::init_from_binary_frm_image(THD *thd, bool write,
                                        extra2.field_data_type_info))
     goto err;
+  /*
+    Extract column definitions from the .frm image buffer.
```
1. It's obvious that it happens from .frm image buffer. but it's a valuable addition, that this loop begins column definitions extraction. I'd suggest re-wording it as "Column definitions extraction begins here."
2. Maybe add the symmetric comment at where the extraction ends? May be useful as well!

Krishnateja4367 replied:
Hi @DaveGosselin-MariaDB , @FooBarrior , @svoj 

Thanks for the feedback! I've updated the comment to use a proper C-style block, reworded it for clarity, repositioned it to align with the start of the loop, and added a matching end comment. Let me know if anything else needs adjustment.
Appreciate your guidance,
Krishna Teja.

Krishnateja4367 commented:
> @cvicentiu what is the status of #2191? Should we stick to #2191 and try to get it completed, or is this task good enough for another gsoc round?

I’d be happy to explore issue [#2191](https://github.com/MariaDB/server/issues/2191) if there’s scope to contribute or improve it further. I’d love to help complete it if it aligns with GSoC goals.

Krishnateja4367 commented:
> Please squash the commits to one commit and update the PR.

I updated the PR

DaveGosselin-MariaDB commented:
> > Please squash the commits to one commit and update the PR.
> 
> I updated the PR

Thanks.  Please rebase to the latest `main` as there's a fix for the `timezone` test there.

Krishnateja4367 commented:
> > > Please squash the commits to one commit and update the PR.
> > 
> > 
> > I updated the PR
> 
> Thanks. Please rebase to the latest `main` as there's a fix for the `timezone` test there.

Hi, I’ve rebased my branch and made the necessary changes for PR #3938, but I’m unable to see the “Merge pull request” button. It shows 0 checks completed. Could you please assist or let me know if there’s any other action I need to take?
